src/mono/netcore/System.Private.CoreLib/src/System/Threading/ThreadPool.Browser.Mono.cs

```diff
@@ -0,0 +1,120 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.Threading
+{
+    public sealed class RegisteredWaitHandle : MarshalByRefObject
+    {
+        internal RegisteredWaitHandle(WaitHandle waitHandle, _ThreadPoolWaitOrTimerCallback callbackHelper,
+            int millisecondsTimeout, bool repeating)
+        {
+        }
+
+        public bool Unregister(WaitHandle? waitObject)
+        {
+            throw new PlatformNotSupportedException();
+        }
+    }
+
+    public static partial class ThreadPool
+    {
+        internal const bool EnableWorkerTracking = false;
+
+        private static bool _callbackQueued;
+
+        internal static void InitializeForThreadPoolThread() { }
+
+        public static bool SetMaxThreads(int workerThreads, int completionPortThreads)
+        {
+            if (workerThreads == 1 || completionPortThreads == 1)
+                return true;
+            return false;
+        }
+
+        public static void GetMaxThreads(out int workerThreads, out int completionPortThreads)
+        {
+            workerThreads = 1;
+            completionPortThreads = 1;
+        }
+
+        public static bool SetMinThreads(int workerThreads, int completionPortThreads)
+        {
+            if (workerThreads == 1 || completionPortThreads == 1)
```

Comment:
Should this be && instead of || ?  As written it means we'll return true for SetMinThreads(int.MaxValue, 1).

Reply:
I'm not really sure how these methods should behave. Should we act like wasm has 0 threads, or it has 1 ? Same for the completion port threads.

Reply:
Yeah. it's a weird situation; nothing we do here will entirely make sense. We effectively have 1 thread that serves all purposes, so I think it makes the most sense to say that the min and max for both values is 1.  We could track what the user set it to and roundtrip it, e.g. if they set it to 0 then return 0 when getting it, but it's probably not worth it to do so.

@kouvel, opinions?

Reply:
Currently in coreclr SetMinThreads allows values of 0 but it will store and return 1 from GetMinThreads in that case, not sure why, guess it was like that before. SetMaxThreads allows values > 0.

For this case I suppose it could allow only 1 for both values in Set* and return 1 for both in Get*. Some of the current tests may need to be disabled in this mode.